src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx

```diff
@@ -39,40 +57,73 @@ export const ConfigureTasks: React.FC<Props> = ({
   );
 
   const tasks = selectedBuildVariants.map(
-    (bv) => selectedBuildVariantTasks[bv]
+    (bv) => selectedBuildVariantTasks[bv] || {}
   );
 
   const currentTasks = deduplicateTasks(tasks);
+  const currentDownstreamPatches = getVisibleDownstreamPatches(
+    selectedDownstreamPatches,
+    selectedBuildVariants
+  );
+  const currentChildPatches = getVisibleChildPatches(
+    childPatches,
+    selectedBuildVariants
+  );
+
+  // Show details related to child patches (i.e. list all variants/tasks) only if it is the only menu item selected
+  const enumerateChildPatches =
+    currentChildPatches.length === 1 && selectedBuildVariants.length === 1;
 
   const onClickCheckbox = (taskName: string) => (e) => {
     const selectedBuildVariantsCopy = { ...selectedBuildVariantTasks };
-    selectedBuildVariants.forEach((bv) => {
-      if (selectedBuildVariantsCopy[bv][taskName] !== undefined) {
-        selectedBuildVariantsCopy[bv][taskName] = e.target.checked;
+    const selectedDownstreamPatchesCopy = { ...selectedDownstreamPatches };
+    selectedBuildVariants.forEach((v) => {
```

Comment:
Good question, this is [existing code](https://github.com/evergreen-ci/spruce/blob/32ce975c828926b398d9ba0cac1b287b2d6aaa5e/src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx#L49) that I just moved into the hook. I would imagine that the goal here is to iterate over an unchanging object and avoiding the messiness of making updates to the same object that's being looped. Since it's working fine I'm inclined to keep as-is!